lib/active_merchant/billing/gateways/hi_pay.rb

```diff
@@ -51,23 +51,42 @@ def authorize(money, payment_method, options = {})
       end
 
       def capture(money, authorization, options)
-        post = {}
-        post[:operation] = 'capture'
-        post[:currency] = (options[:currency] || currency(money))
-        transaction_ref, _card_token, _payment_product = authorization.split('|')
-        commit('capture', post, { transaction_reference: transaction_ref })
+        maintenance_operation(money, authorization, options.merge({ operation: 'capture' }))
       end
 
       def store(payment_method, options = {})
         tokenize(payment_method, options.merge({ multiuse: '1' }))
       end
 
-      def scrub(transcrip)
-        # code
+      def refund(money, authorization, options)
+        maintenance_operation(money, authorization, options.merge({ operation: 'refund' }))
+      end
+
+      def void(authorization, options)
+        maintenance_operation(nil, authorization, options.merge({ operation: 'cancel' }))
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Basic )[\w =]+), '\1[FILTERED]').
+          gsub(%r((card_number=)\w+), '\1[FILTERED]\2').
+          gsub(%r((cvc=)\w+), '\1[FILTERED]\2')
       end
 
       private
 
+      def maintenance_operation(money, authorization, options)
```

Comment:
Can we rename this to be something a bit more specific. Maybe `reference_operation`?

Reply:
In this case, the name was according to those operations' names in the documentation. https://developer.hipay.com/api-explorer/api-online-payments#/maintenance/performMaintenanceOperation.

But yes, I can rename it.